src/main/kotlin/depromeet/hotsix/obrit/receipt/client/ReceiptOcrClient.kt

```
@@ -39,27 +39,31 @@ class ReceiptOcrClient(
+                ?: throw OcrFailedException(OcrFailureReason.EMPTY_RESPONSE, "API 응답에서 분석 결과를 찾을 수 없습니다.")
 
             objectMapper.readValue(jsonText, OcrAnalysisResponse::class.java)
         } catch (e: HttpServerErrorException) {
             // 5xx 서버 에러 (503 등)
-            throw BusinessException(
+            throw OcrFailedException(
+                OcrFailureReason.UPSTREAM_5XX,
                 "AI API 서버 오류 (${e.statusCode}): 잠시 후 다시 시도해주세요.",
             )
-        } catch (e: BusinessException) {
+        } catch (e: ResourceAccessException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)